framework/src/source/TestRunner.bs

```diff
@@ -103,7 +103,9 @@ namespace rooibos
 
       m.stats.time = rooibosTimer.totalMilliseconds()
 
-      m.testReporter.reportResults(m.stats)
+      for each reporter in m.testReporters
+        reporter.onEnd(m.stats)
```

Comment:
None of the reporter hooks are required, so we need to guard this with an "is onEnd defined" check before calling it.

Reply:
Let's pass in stats like this (so it conforms to our `event` object structure that we haven't flushed out yet)

```
{ stats: m.stats }
```

Reply:
Done.